From 24b759c5d79c1a4993c05c1ef7f44f9ff6d7f463 Mon Sep 17 00:00:00 2001
From: David Herrmann <dh.herrmann@gmail.com>
Date: Sun, 11 Jan 2015 03:13:46 +0100
Subject: [PATCH] log: fix log_full_errno() with custom facilities

Make sure to extract the log-priority when comparing against
max-log-level, otherwise, we will always drop those messages.

This fixes bus-proxyd to properly send warnings on policy blocks.
---
 src/shared/log.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/log.h b/src/shared/log.h
index 2b6971f..d15d7c8 100644
--- a/src/shared/log.h
+++ b/src/shared/log.h
@@ -158,7 +158,7 @@ void log_assert_failed_return(
 #define log_full_errno(level, error, ...)                                         \
         ({                                                                        \
                 int _l = (level), _e = (error);                                   \
-                (log_get_max_level() >= _l)                                       \
+                (log_get_max_level() >= LOG_PRI(_l))                              \
                 ? log_internal(_l, _e, __FILE__, __LINE__, __func__, __VA_ARGS__) \
                 : -abs(_e); \
         })
-- 
2.3.0

